lib/mongoid/association/relatable.rb

```diff
@@ -411,7 +411,7 @@ def default_inverse
       def namespace_hierarchy(mod)
         parent = Object
         hier = [parent]
-        mod.name.split('::').each do |part|
+        mod.name&.split('::')&.each do |part|
```

Comment:
Can you please change this to 

```
if mod.name
  ...
end
```

instead of try-traversing the entire chain.

Reply:
+ add a comment that `mod.name` is nil for anonymous classes, and for those the hierarchy is just `[Object]`.

Reply:
Both above done.